### PR TITLE
fix recusrive submods

### PR DESCRIPTION
**Pull request by tnichols217**
Fix Issue #216

### CODE REVIEW

File: src/simpleGit.ts [hunk 1]
```diff
@@ -90,7 +90,36 @@ export class SimpleGit extends GitManager {
     async commitAll(message: string): Promise<number> {
         if (this.plugin.settings.updateSubmodules) {
             this.plugin.setState(PluginState.commit);
-            await this.git.subModule(["foreach", "--recursive", `git add -A && if [ ! -z "$(git status --porcelain)" ]; then git commit -m "${await this.formatCommitMessage(message)}"; fi`], (err) => this.onError(err));
+            this.git.outputHandler(async (x, y, z) => {
+                let body = ""
+                let root = this.app.vault.adapter.basePath
```
Comment (Vinzent03) [inline src/simpleGit.ts:95]: `root` is since #209 not just the basepath

File: src/simpleGit.ts [hunk 2]
```diff
@@ -90,7 +90,36 @@ export class SimpleGit extends GitManager {
...
+                y.on('data', (chunk) => {
+                    body += chunk.toString('utf8')
+                })
+                y.on('end', async () => {
+                    let m = body.split('\n')
+                    let l = m.map(x => {
+                        let a = x.match(/'([^']*)'/)
+                        if (a != undefined) {
+                            if (x.startsWith("Entering")) {
```
Comment (Vinzent03) [inline src/simpleGit.ts:104]: I don't know, but what about other languages?

**Review by Vinzent03 — CHANGES_REQUESTED** [review]
Could you please explain your changes? I left already 2 basic comments.

File: src/simpleGit.ts (hunk 1 above)
Comment (tnichols217) [reply src/simpleGit.ts]: Oh, my bad, ill fix that now

File: src/simpleGit.ts (hunk 2 above)
Comment (tnichols217) [reply src/simpleGit.ts]: Ah, that was an uneccessary check i did, removed it

**Comment by tnichols217** [comment]
All i do is switch the command from `git submodule --recursive foreach` to just `git submodule --recursive` in order to parse the stdout to get the list of submodules, which i then parse to get just the directory, enter those directories and commit and push

**Comment by Vinzent03** [comment]
@tnichols217 Okay thanks for the explanation. Can you please choose some proper variable names instead of `x,y,z,m,l`? It's hard to read.

**Comment by tnichols217** [comment]
> @tnichols217 Okay thanks for the explanation. Can you please choose some proper variable names instead of `x,y,z,m,l`? It's hard to read.

Apologies, it's a habit from competitive programming. I'll change that now.

**Comment by tnichols217** [comment]
Done! Hows this?

**Comment by Vinzent03** [comment]
Hey I just added my own changes. I tried to properly await the commits inside the handler and some styles. Could you please verify that it's still working on your end?

**Comment by tnichols217** [comment]
> Hey I just added my own changes. I tried to properly await the commits inside the handler and some styles. Could you please verify that it's still working on your end?

Just tested, working perfect. Thanks for the feedback!

**Review by Vinzent03 — APPROVED** [review]

**Pull request merged**